upload/views/test_results.py

```diff
@@ -0,0 +1,116 @@
+    build = serializers.CharField(required=False)
+    buildURL = serializers.CharField(required=False)
+    job = serializers.CharField(required=False)
+    flags = FlagListField(required=False)
+    pr = serializers.CharField(required=False)
+    service = serializers.CharField(required=False)
+
+
+class TestResultsView(APIView):
+    permission_classes = [UploadTestResultsPermission]
+    authentication_classes = [
+        OrgLevelTokenAuthentication,
+        RepositoryLegacyTokenAuthentication,
+    ]
+
+    def post(self, request):
+        serializer = UploadSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        data = serializer.validated_data
+
+        if isinstance(request.user, Owner):
+            # using org token
+            owner = request.user
+            repo = get_repository_from_string(Service(owner.service), data["slug"])
+        elif isinstance(request.user, RepositoryAsUser):
+            # repository token
+            repo = request.user._repository
+        else:
+            raise NotAuthenticated()
+
+        commit, _ = Commit.objects.get_or_create(
+            commitid=data["commit"],
+            repository=repo,
+            defaults={
+                "branch": data.get("branch"),
+                "pullid": data.get("pr"),
+                "merged": False if data.get("pr") is not None else None,
+                "state": "pending",
+            },
+        )
+
+        upload_external_id = str(uuid.uuid4())
+
+        storage_path = f"v1/uploads/{upload_external_id}.json"
+
+        archive_service = ArchiveService(repo)
+        url = archive_service.create_presigned_put(storage_path)
+
+        task_arguments = {
+            # these are used in the upload task when saving an upload record
+            # and use some unfortunately named and confusing keys
+            # (eventual reports_upload columns indicated by comments)
+            "reportid": upload_external_id,  # external_id
+            "build": data.get("build"),  # build_code
+            "build_url": data.get("buildURL"),  # build_url
+            "job": data.get("job"),  # job_code
+            "flags": data.get("flags"),
+            "service": data.get("service"),  # provider
+            "url": storage_path,  # storage_path
+            # these are used for dispatching the task below
+            "commit": commit.commitid,
+            "report_code": None,
+        }
+
+        log.info(
+            "Dispatching test results upload to worker",
+            extra=dict(
+                commit=commit.commitid,
+                repoid=repo.repoid,
+                task_arguments=task_arguments,
+            ),
+        )
+
+        dispatch_upload_task(
+            task_arguments,
+            repo,
+            get_redis_connection(),
+            report_type=CommitReport.ReportType.TEST_RESULTS,
```

Comment:
I noticed that in `dispatch_upload_task` the BUNDLE_ANALYSIS has a delay of 4 ([here](https://github.com/codecov/codecov-api/blob/a9a1f250afb24ed0900b26ff47da5fc5182ea854/upload/helpers.py#L674)), same for coverage reports with version v4.

I believe this delay tells celery "wait 4s to enqueue this task". This is because the user's CI needs time to upload the report to GCS before we can actually process it. Adding the delay (I think) reduces the chance of errors of the kind "we can't process because the upload was not found in storage".

I guess it seems prudent to follow suit and add this delay as well? I suppose it depends on how big the test result reports can be... totally up to you to add or not such delay, but I thought best to point this out anyway.